### PR TITLE
6730 date range picker

### PR DESCRIPTION
**Pull request by KombiCode**
#### What? Why?

Ongoing # 6730

Use range datepicker for orders and bulk order management

#### What should we test?
Orders : visit url /admin/orders, choose a range within the date picker and click on Filter Results to see if results match the range
Bulk Order Management : visit url /admin/orders/bulk_management, choose a range within the date picker and see if corresponding orders are displayed

#### Release notes

Next step is to modify datepicker for subscription and for reports

### CODE REVIEW

File: app/assets/javascripts/admin/util.js.erb [hunk 1]
```diff
@@ -18,6 +18,7 @@ $(document).ready(function() {
     }
   }
   window.FLATPICKR_DATE_DEFAULT = {
+    mode: "range",
```
Comment (jibees) [inline app/assets/javascripts/admin/util.js.erb:21]: Not all flatpickr instances will use this mode. We should add this options only where it's relevant.

**Comment by RachL** [comment]
@jibees @KombiCode as this is still a draft, I'm moving it back to in dev instead of code review. Let me know if this is wrong.

**Comment by KombiCode** [comment]
@jibees I made some trys with use of rangeplugin instead of mode range.
Indeed it should be easier for data management.
Since with this plugin there are two inputs, for presentation we could style to have something like below :
Capture d’écran 2022-02-15 à 11 05 36

During the input it would look like :
Capture d’écran 2022-02-15 à 11 07 28

And once input finished, it would look like :
Capture d’écran 2022-02-15 à 11 07 56

Does it sound good ?

**Comment by jibees** [comment]
Sounds good to me. I'm a bit concerned about the "End date" alignment: should this label be aligned on the right? 

I'd invoke @openfoodfoundation/train-drivers-product-owners

**Comment by KombiCode** [comment]
With the range plugin the flatpickr controls two inputs and the behavior I saw so far is that one can click on each input...
Here we 'gather' visually the two inputs (whereas in the example of [https://flatpickr.js.org/plugins/#rangeplugin-beta](url) the two inputs are distinct)
I agree it would be better UX if user could click only on 'start date' area : need to see if this could be done.
This range plugin pro is the simplification to retrieve start and end separately for data storage, con is the display
And vice-versa for the 'mode range'

**Comment by RachL** [comment]
@KombiCode I know you had a call with @jibees on this but I guess you didn't have time to get back to this PR. So I'm going to close it, don't hesitate to reopen if you do work on it again.

**Comment by KombiCode** [comment]
@RachL, indeed already more than two months since I discussed this with @jibees. At that time I thought I could finish that work but due to some personal events I finally could not. Currently I'm very busy with other stuff and you did well to close the PR. I'm really sorry to have not accomplished the work on that.

**Comment by RachL** [comment]
No worries @KombiCode ;) Good luck on your other work!

**Comment by KombiCode** [comment]
Thanks @RachL

**Pull request closed without merging**